enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/FileUserRealm.java

```diff
@@ -106,9 +108,15 @@ public FileUserRealm( UserRepository userRepository, RoleRepository roleReposito
     protected AuthorizationInfo doGetAuthorizationInfo( PrincipalCollection principals )
     {
         User user = userRepository.findByName( (String) principals.getPrimaryPrincipal() );
-
-        Set<String> roles = roleRepository.findByUsername( user.name() );
-        return new SimpleAuthorizationInfo( roles );
+        if ( user.passwordChangeRequired() || user.hasFlag( IS_SUSPENDED ))
```

Comment:
Have `user.isSuspended()` instead of checking the flag here so as not to leak implementation details.